internal/protocol/gen/gen.go

```diff
@@ -0,0 +1,59 @@
+// +build generate
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+const (
+	goplsRef = "gopls/v0.5.3"
+	urlFmt   = "https://raw.githubusercontent.com/golang/tools" +
+		"/%s/internal/lsp/protocol/tsprotocol.go"
+)
+
+func main() {
+	args := os.Args[1:]
+	if len(args) > 1 && args[0] == "--" {
+		args = args[1:]
+	}
+
+	if len(args) != 1 {
+		log.Fatalf("expected exactly 1 argument (target path), given: %q", args)
+	}
+
+	filename, err := filepath.Abs(args[0])
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	url := fmt.Sprintf(urlFmt, goplsRef)
+
+	resp, err := http.Get(url)
```

Comment:
This is far simpler way of extracting it compared to [govim's approach](https://github.com/govim/govim/tree/main/cmd/govim/internal/golang_org_x_tools), which we can apply just because we're only interested in the protocol structs and nothing else and thankfully everything is contained within that one file and that file does not have any non-stdlib imports.

It may need to be revisited if/when we get closer to extracting the JSON-RPC package before it's exported.